packages/react-components/react-progress/src/components/Progress/useProgressStyles.ts

```diff
@@ -1,33 +1,201 @@
+      ${tokens.colorNeutralBackground6} 100%
+    )`,
+    animationName: IndeterminateProgress,
+    animationDuration: '3s',
+    animationIterationCount: 'infinite',
+  },
+
+  rtl: {
+    animationDirection: 'reverse',
   },
+});
+
+const useTrackStyles = makeStyles({
+  base: {
+    gridRowStart: '2',
+    gridColumnStart: '1',
+    backgroundColor: tokens.colorNeutralBackground6,
 
-  // TODO add additional classes for different states and/or slots
+    '@media screen and (forced-colors: active)': {
+      ...shorthands.borderBottom('1px', 'solid', 'CanvasText'),
+    },
+  },
+  medium: {
+    height: barThicknessValues.medium,
+  },
+  large: {
+    height: barThicknessValues.large,
+  },
 });
 
 /**
  * Apply styling to the Progress slots based on the state
  */
 export const useProgressStyles_unstable = (state: ProgressState): ProgressState => {
-  const styles = useStyles();
-  state.root.className = mergeClasses(progressClassNames.root, styles.root, state.root.className);
+  const { indeterminate = false, thickness = 'medium', percentComplete = 0 } = state;
+  const rootStyles = useRootStyles();
+  const barStyles = useBarStyles();
+  const trackStyles = useTrackStyles();
+  const labelStyles = useLabelStyles();
+  const descriptionStyles = useDescriptionStyles();
+  const { dir } = useFluent();
+
+  const valuePercent = !indeterminate ? Math.min(100, Math.max(0, percentComplete! * 100)) : undefined;
+  const progressBarStyles = {
+    [`${progressCssVars.percentageCssVar}`]: !indeterminate ? valuePercent + '%' : undefined,
+  };
+
+  state.root.className = mergeClasses(progressClassNames.root, rootStyles.root, state.root.className);
+
+  // Determinate indicator when determinate is defined
+  if (state.bar) {
+    state.bar.className = mergeClasses(
+      progressClassNames.bar,
+      barStyles.base,
+      indeterminate && barStyles.indeterminate,
+      indeterminate && barStyles[dir],
+      barStyles[thickness],
+      !indeterminate && barStyles.determinate,
+      !indeterminate && !!valuePercent && valuePercent > ZERO_THRESHOLD && barStyles.nonZeroDeterminate,
+      state.bar.className,
+    );
+  }
+
+  if (state.track) {
+    state.track.className = mergeClasses(
+      progressClassNames.track,
+      trackStyles.base,
+      trackStyles[thickness],
+      state.track.className,
+    );
+  }
+
+  if (state.label) {
+    state.label.className = mergeClasses(progressClassNames.label, labelStyles.default, state.label.className);
+  }
+
+  if (state.description) {
+    state.description.className = mergeClasses(
+      progressClassNames.description,
+      descriptionStyles.default,
+      state.description.className,
+    );
+  }
 
-  // TODO Add class names to slots, for example:
-  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);
+  if (state.bar && percentComplete) {
+    state.bar.style = {
+      ...progressBarStyles,
+      ...state.bar.style,
+    };
+  }
```

Comment:
I think you want to check `!indeterminate` instead of `percentComplete` right? Also since `progressBarStyles` only ever has one element, you could just define it inline here.

```suggestion
  if (state.bar && !indeterminate) {
    state.bar.style = {
      [progressCssVars.percentageCssVar]:  Math.min(100, Math.max(0, percentComplete * 100)) + '%',
      ...state.bar.style,
    };
  }
```